orderer/common/cluster/block_fetcher.go

```diff
@@ -0,0 +1,503 @@
+/*
+Copyright IBM Corp. 2022 All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster
+
+import (
+	"context"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	ab "github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/util"
+	"github.com/hyperledger/fabric/internal/pkg/identity"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
```

Comment:
general comment - i think we should have much more logging in this file.

Reply:
taken